torchgeo/datasets/cloud_cover.py

```diff
@@ -0,0 +1,401 @@
+            os.path.split(item_json)[0], label_data["assets"]["labels"]["href"]
+        )
+        item_meta["target"] = [label_asset_path]
+
+        source_item_hrefs = []
+        for link in label_data["links"]:
+            if link["rel"] == "source":
+                source_item_hrefs.append(
+                    os.path.join(self.root, link["href"].replace("../../", ""))
+                )
+
+        source_item_hrefs = sorted(source_item_hrefs)
+        source_item_paths = []
+
+        for item_href in source_item_hrefs:
+            source_item_path = os.path.split(item_href)[0]
+            source_data = self._read_json_data(item_href)
+            source_item_assets = []
+            for asset_key, asset_value in source_data["assets"].items():
+                if asset_key in self.bands:
+                    source_item_assets.append(
+                        os.path.join(source_item_path, asset_value["href"])
+                    )
+            source_item_assets = sorted(source_item_assets)
+            for source_item_asset in source_item_assets:
+                source_item_paths.append(source_item_asset)
+
+        item_meta["source"] = source_item_paths
+        return item_meta
+
+    def _load_collections(self) -> List[Dict[str, Any]]:
+        """Loads the paths to source and label assets for each collection.
+
+        Returns:
+            a dictionary with lists of filepaths to all assets for each chip/item
+
+        Raises:
+            RuntimeError if collection.json is not found in the uncompressed dataset
+        """
+        indexed_chips = []
+        label_collection: List[str] = []
+        for c in self.collection_names[self.split]:
+            if "label" in c:
+                label_collection.append(c)
+        label_collection_path = os.path.join(self.root, label_collection[0])
+        label_collection_json = os.path.join(label_collection_path, "collection.json")
+
+        label_collection_item_hrefs = []
+        for link in self._read_json_data(label_collection_json)["links"]:
+            if link["rel"] == "item":
+                label_collection_item_hrefs.append(link["href"])
+
+        label_collection_item_hrefs = sorted(label_collection_item_hrefs)
+
+        for label_href in label_collection_item_hrefs:
+            label_json = os.path.join(label_collection_path, label_href)
+            indexed_item = self._load_items(label_json)
+            indexed_chips.append(indexed_item)
+
+        return indexed_chips
+
+    def _validate_bands(self, bands: Sequence[str]) -> None:
+        """Validate list of bands.
+
+        Args:
+            bands: user-provided tuple of bands to load
+
+        Raises:
+            ValueError: if an invalid band name is provided
+        """
+        for band in bands:
+            if band not in self.band_names:
+                raise ValueError(f"'{band}' is an invalid band name.")
+
+    def _check_integrity(self) -> bool:
```

Comment:
This is an optional request. I know you've already spent a ton of time on this PR, and I would rather it be merged as is than have you give up. With that said:

Our older datasets use `_check_integrity` and `_download`, but have a problem. If you have already downloaded the dataset but not yet extracted it, or if you have extracted it and deleted the tar/zip files, the dataset will try to re-download and re-extract everything. All of our newer datasets now have separate `_verify`, `_download`, and `_extract` methods to properly support this. We would like to move all of our datasets to the latter format at some point but haven't gotten around to it yet. 

So if you want to convert this dataset to the new format, that would help. But if you want to save that for another day or another person, I totally understand.